src/microbe_stage/PatchMap.cs

```diff
@@ -320,4 +370,127 @@ public void ReplaceSpecies(Species old, Species newSpecies)
             patch.ReplaceSpecies(old, newSpecies);
         }
     }
+
+    public void SetSeed(Random random)
+    {
+        if (Seed != null)
+            throw new ArgumentException("cant set seed, seed has already been set");
+
+        Seed = random;
+    }
+
+    public void BuildSpecialRegions()
+    {
+        foreach (var region in SpecialRegions)
+        {
+            region.Value.BuildRegion();
+        }
+    }
+
+    public void BuildPatchesInRegions()
+    {
+        foreach (var region in Regions)
+        {
+            region.Value.BuildPatches(Seed);
+            foreach (var patch in region.Value.Patches)
+            {
+                AddPatch(patch);
+            }
+        }
+    }
+
+    public void BuildPatchesInSpecialRegions()
+    {
+        foreach (var region in SpecialRegions)
+        {
+            region.Value.BuildPatches(Seed);
+            foreach (var patch in region.Value.Patches)
+            {
+                AddPatch(patch);
+            }
+        }
+    }
+
+    public void ConnectPatchesBetweenRegions()
+    {
+        foreach (var region in Regions)
+        {
+            region.Value.ConnectPatchesBetweenRegions(Seed);
+        }
+    }
+
+    public bool ContainsPatchAdjacency(int id1, int id2)
```

Comment:
This should have a comment / the name should be adjusted to tell that this looks for links in both directions. It's technically possible to have one direction links which is a feature I'll fight to keep possible code-wise for the future.